#VRML_SIM R2021b utf8
WorldInfo {
  coordinateSystem "NUE"
}
Viewpoint {
  orientation -0.9998528965008924 -0.012632188215276683 -0.011602292000705464 1.4481073953941312
  position -0.0008591000011925476 2.9294096715220976 0.49576897642636386
}
TexturedBackground {
}
TexturedBackgroundLight {
}
DEF area RectangleArena {
  translation 0.00608592 -4.44089e-16 -0.0128875
  floorSize 1.8288 1.8288
  floorTileSize 0.9144 0.9144
}
WoodenBox {
  translation 0.466039 2.39808e-14 -0.020027
  rotation 0 1 0 -1.5707953071795862
  size 0.9144 0.2 0.01
}
WoodenBox {
  translation -0.437759 4.08562e-14 -0.0141464
  rotation 0 1 0 -1.5707953071795862
  name "wooden box(3)"
  size 0.9144 0.2 0.01
}
WoodenBox {
  translation 0.0111685 4.04121e-14 0.43907
  name "wooden box(1)"
  size 0.9144 0.2 0.01
}
WoodenBox {
  translation 0.0114054 8.88176e-15 -0.476269
  name "wooden box(2)"
  size 0.9144 0.2 0.01
}
DEF epuck E-puck {
  translation 0.715619 -3.94552e-06 0.676571
  rotation -4.9580396042433186e-06 -0.9999999201788042 0.0003995219681096762 -1.5707853071795865
  controller "Lab2_Task3_CorridorLeftTurns"
  supervisor TRUE
  camera_width 50
  turretSlot [
    InertialUnit {
      rotation 0 1 0 3.14
    }
    DEF righthinge HingeJoint {
      jointParameters HingeJointParameters {
        axis -1 0 0
        anchor 0 0.025 0
      }
      device [
        RotationalMotor {
          name "right wheel motor"
          consumptionFactor 70
          controlPID 8 2 0.5
        }
        PositionSensor {
          name "right wheel sensor"
        }
      ]
      endPoint Solid {
        translation 0.045009957783801 0.025013380758883635 -2.4671929852070235e-05
        rotation 0.9999993203499804 0.0010052734270078172 -0.000590529350852493 4.6368371220546685
        children [
          DEF WHEEL Transform {
            translation -0.007 0.01 -0.05
            rotation 0 0 1 1.57
            children [
              Shape {
                appearance PBRAppearance {
                  baseColor 0 0 0
                  transparency 1
                  roughness 1
                  metalness 0
                }
                geometry Cylinder {
                  height 0.01
                  radius 0.02
                }
              }
            ]
            rotationStep 0.25
          }
        ]
        name "right wheel(1)"
        boundingObject USE WHEEL
        physics DEF PHYSICS_WHEEL Physics {
          density -1
          mass 0.05
        }
        rotationStep 0.25
      }
    }
    DEF lefthinge HingeJoint {
      jointParameters HingeJointParameters {
        axis -1 0 0
        anchor 0 0.025 0
      }
      device [
        RotationalMotor {
          name "left wheel motor"
          consumptionFactor 70
          controlPID 8 0 0
        }
        PositionSensor {
          name "left wheel sensor"
        }
      ]
      endPoint Solid {
        translation -0.04501012942703837 0.024977671977279067 -4.5733976162035045e-05
        rotation -0.9999982113395169 0.00047275268613733115 -0.0018313444964861228 1.2425810202889571
        children [
          DEF WHEEL Transform {
            translation 0.007 0 -0.05
            rotation 0 0 1 1.57
            children [
              Shape {
                appearance PBRAppearance {
                  baseColor 0.1 0.1 0.1
                  transparency 1
                  roughness 1
                  metalness 0
                }
                geometry Cylinder {
                  height 0.01
                  radius 0.02
                }
              }
            ]
            rotationStep 0.25
          }
        ]
        name "left wheel(1)"
        boundingObject USE WHEEL
        physics DEF PHYSICS_WHEEL Physics {
          density -1
          mass 0.05
        }
      }
    }
    DEF front_ds DistanceSensor {
      translation 0 0 -0.045
      rotation 0 1 0 1.5708
      name "front_ds"
      recognitionColors [
        0 0 0
      ]
      lookupTable [
        0 0 0
        1.27 1.27 0
      ]
    }
    DEF right_ds DistanceSensor {
      translation 0.035 -0.01 -0.01
      name "right_ds"
      lookupTable [
        0 0 0
        1.27 1.27 0
      ]
    }
    DEF left_ds DistanceSensor {
      translation -0.035 -0.01 -0.01
      rotation 0 1 0 3.14159
      name "left_ds"
      lookupTable [
        0 0 0
        1.27 1.27 0
      ]
    }
    Camera {
      translation 0 -0.01 -0.045
      name "camera1"
      recognitionColors [
        1 1 0
      ]
      fieldOfView 1.57
      width 80
      height 80
      recognition Recognition {
      }
    }
  ]
}
